src/rtc.rs

```diff
@@ -0,0 +1,849 @@
+        if self.reg.cr.read().bkp().bit_is_set() {
+            DstState::Dst
+        } else {
+            DstState::Standard
+        }
+    }
+
+    /// Sets the stored Daylight Savings Time status without adjusting the clock
+    pub fn set_dst(&mut self, dst: DstState) {
+        self.reg.cr.modify(|_, w| w.bkp().bit(dst == DstState::Dst));
+    }
+
+    /// Begin Daylight Savings Time
+    ///
+    /// Adds an hour to the stored time and sets the stored DST status.
+    /// Returns an error if the clock has not been initialized, or if the stored DST status
+    /// indicates it has already begun.
+    pub fn begin_dst(&mut self) -> Result<(), DstError> {
+        self.calendar_initialized()
+            .ok_or(DstError::ClockNotInitialized)?;
+        if self.reg.cr.read().bkp().bit_is_set() {
+            return Err(DstError::AlreadyDst);
+        }
+        self.reg
+            .cr
+            .modify(|_, w| w.add1h().set_bit().bkp().set_bit());
+        Ok(())
+    }
+
+    /// End Daylight Savings Time
+    ///
+    /// Subtracts an hour from the stored time and sets the stored DST status.
+    /// Returns an error if the clock has not been initialized, if the stored DST status
+    /// indicates it is already standard time, or if we cannot subtract an hour because
+    /// it is not at least one hour past midnight.
+    pub fn end_dst(&mut self) -> Result<(), DstError> {
+        self.calendar_initialized()
+            .ok_or(DstError::ClockNotInitialized)?;
+        if self.reg.cr.read().bkp().bit_is_clear() {
+            return Err(DstError::AlreadyStandardTime);
+        }
+        let time = self.reg.tr.read();
+        if time.ht().bits() == 0 && time.hu().bits() == 0 {
+            return Err(DstError::CannotSubtract);
+        }
+        self.reg
+            .cr
+            .modify(|_, w| w.sub1h().set_bit().bkp().clear_bit());
+        Ok(())
+    }
+
+    /// Configures the wakeup timer to trigger periodically after `interval` seconds
+    ///
+    /// # Panics
+    ///
+    /// Panics if interval is greater than 2¹⁷-1.
+    pub fn enable_wakeup(&mut self, interval: u32) {
+        self.reg.cr.modify(|_, w| w.wute().clear_bit());
+        self.reg.isr.modify(|_, w| w.wutf().clear_bit());
+        while self.reg.isr.read().wutwf().bit_is_clear() {}
+
+        if interval > 1 << 16 {
+            self.reg
+                .cr
+                .modify(|_, w| unsafe { w.wucksel().bits(0b110) });
+            let interval = u16(interval - 1 << 16)
+                .expect("Interval was too large for wakeup timer");
+            self.reg.wutr.write(|w| unsafe { w.wut().bits(interval) });
+        } else {
+            self.reg
+                .cr
+                .modify(|_, w| unsafe { w.wucksel().bits(0b100) });
+            let interval =
+                u16(interval).expect("Interval was too large for wakeup timer");
+            self.reg.wutr.write(|w| unsafe { w.wut().bits(interval) });
```

Comment:
Section 49.6.6
> the WUTF flag is set every (WUT[15:0] + 1) ck_wut cycles

I think we want to set the flag every `interval` cycles, so it needs a - 1 here.

Reply:
That would explain why the example woke up every 11 seconds... ;)